full.aws/src/full/aws/sqs.clj

```diff
@@ -0,0 +1,309 @@
+         (string? receipt-handle)]}
+  (thread-try
+    (->> (DeleteMessageRequest. queue-url receipt-handle)
+         (.deleteMessage @client))))
+
+(defn change-message-visibility>
+  [{:keys [queue-url receipt-handle] :as message}
+   visibility-timeout]
+  {:pre [(string? queue-url)
+         (string? receipt-handle)
+         (integer? visibility-timeout)]}
+  (thread-try
+    (->> (ChangeMessageVisibilityRequest. queue-url
+                                          receipt-handle
+                                          (int visibility-timeout))
+         (.changeMessageVisibility @client))))
+
+(defn- handle-message>
+  [message handler> retries]
+  (go
+    (try
+      (<? (handler> message))
+      (catch Throwable ex
+        (let [retry-message (if retries
+                              (let [receive-count (:approximate-receive-count
+                                                    (:attributes message))
+                                    delay (or (get retries (dec receive-count))
+                                              (last retries))
+                                    ; distribute the messages over a time period
+                                    delay (int (+ (/ delay 2) (rand-int delay)))
+                                    ]
+                                ; Simply make message invisible for the given
+                                ; time.
+                                (<! (change-message-visibility> message
+                                                                delay))
+                                (str "will retry in " delay "sec "
+                                     "(" receive-count ")"))
+                              ; else
+                              "not retrying")]
+          ; else - simply log
+          (log/error (str "Error processing message " message ": " ex
+                          ", " retry-message)))))))
+
+(defn subscribe
+  "Repeatedly fetches messages from SQS queue and invokes handler function for
+   each message with specified parallelism. Automatically retries messages with
+   configurable delay.
+   Parameters:
+   * queue-url  SQS queue URL
+   * handle>    handler function, should accept one parameter (message) and return
+                channel that gets closed when message is processed. If channel
+                yields exception, the message will be retried after a delay
+   * :parallelism  Max number of messages to handle in parallel. Default 1.
+   * :retries      Vector of retry intervals in seconds, for example [4 10] will
+                   mean that first time the message will be retry after
+                   4 +-2 sec, second and following times after 10 +-5 sec."
+  [queue-url handler> & {:keys [visibility-timeout unserializer
+                                parallelism retries]
+                         :or {unserializer read-edn
+                              parallelism 1
+                              retries [5 60 900 3600]}}]
```

Comment:
Iou can override it via function param. Parent app can make it configurable if neccessary I think.